src/amcrest/media.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# vim:sw=4:ts=4:et
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Media(object):
+
+    def factory_create(self):
+        ret = self.command(
+            'mediaFileFind.cgi?action=factory.create'
+        )
+        return ret.content.decode('utf-8')
+
+    def factory_close(self, factory_id):
+        ret = self.command(
+            'mediaFileFind.cgi?action=factory.close&object={0}'
+            .format(factory_id)
+        )
+        return ret.content.decode('utf-8')
+
+    def factory_destroy(self, factory_id):
+        ret = self.command(
+            'mediaFileFind.cgi?action=factory.destroy&object={0}'
+            .format(factory_id)
+        )
+        return ret.content.decode('utf-8')
+
+    def media_file_find_start(self, factory_id,
+                              start_time, end_time, channel=0,
+                              directories=(), types=(), flags=(),
+                              events=(), stream=None):
+        """
+        https://s3.amazonaws.com/amcrest-files/Amcrest+HTTP+API+3.2017.pdf
+
+        factory_id : returned by factory_create()
+
+        dir : in which directories you want to find the file. It is an array.
+                The index starts from 1. The range of dir is {"/mnt/dvr/sda0",
+                "/mnt/dvr/sda1"}. This condition can be omitted. If omitted,
+                find files in all the directories.
+
+        type : which types of the file you want to find. It is an array. The
+                index starts from 0. The range of type is {"dav", "jpg", "mp4"}
+                If omitted, find files with all the types.
+
+        flag : which flags of the file you want to find. It is an array. The
+                index starts from 0. The range of flag is {"Timing", "Manual",
+                "Marker", "Event", "Mosaic", "Cutout"}. If omitted, find files
+                with all the flags.
+
+        event : by which event the record file is triggered. It is an array.
+                The index starts from 0. The range of event is {"AlarmLocal",
+                "VideoMotion", "VideoLoss", "VideoBlind", "Traffic*"}. This
+                condition can be omitted. If omitted, find files of all the
+                events. stream : which video stream type you want to find.
+                The range of stream is {"Main", "Extra1", "Extra2", "Extra3"}.
+                If omitted, find files with all the stream types.
+        """
```

Comment:
Awesome documentation!!!!! :1st_place_medal:

Reply:
Kudos to @jessbo, I've just kept it ;-)

Reply:
@jessbo congrats!